src/main/java/org/folio/circulation/support/ItemRepository.java

```diff
@@ -245,6 +245,10 @@ private ItemRepository(
     HttpResult<MultipleRecords<T>> result,
     BiFunction<T, Item, T> includeItemMap) {
 
+    if (result.value().getRecords().isEmpty()) {
```

Comment:
The result could have failed, which would mean there was no value. 

This could do with a check for failure before it.

I tried to do it elegantly but a way didn’t immediately come to me.

Reply:
I would think "no value" is a condition, not necessarily a failure. For example, not all inventory items having requests or loans. We can check at either multiple places including RequestRepository and LoanRepository or just one place that is here ItemRepository.

Reply:
I think I might have explained myself poorly.

HttpResult’s can either succeed or fail, if they fail, they have no value.

If the result passed into this method failed, then `result.value()` will be null, meaning that `result.getValue().getRecords().isEmpty()` will throw a `NullPointerException`.

I’d like there to be a nice way to achieve this without an explicit check, but I haven’t thought of one yet.

Does that make more sense?

Reply:
@marcjohnson-kint, to be accurate, are you suggesting to add result.failed() check like this?
```
    if (result.failed() || result.value().getRecords().isEmpty()) {
      return CompletableFuture.completedFuture(result);
    }
```

Reply:
looks like our comments crossed at the same time

Reply:
Yeah, I just had to try that out to see if it makes sense.

There should be a way to do this better on the HttpResult class.

Reply:
Let us know when you have a better way :)